Documentation/dev-releases.md

```diff
@@ -59,25 +59,39 @@ git checkout -b v2.1.x tags/v2.1.0
 git push git@github.com:coreos/dex.git v2.1.x
 ```
 
+## Patch releases - cherry pick required commits
+
+If the release is a patch release (2.0.1, 2.0.2, etc.) checkout the desired release branch and cherry pick specific commits.
```

Comment:
Maybe a small note on what kind of changes get into a patch release? e.g. only urgent bug or security fixes.